### PR TITLE
feat(cdk): add valueOf and Symbol.toPrimitive to TuiDay, TuiMonth, TuiYear and TuiTime 

### LINKED ISSUE #1729: 🚀 - Implement valueOf in TuiDay, TuiMonth, TuiYear, TuiTime
### Which @taiga-ui/* package(s) are relevant/releated to the feature request?

cdk

### Description

I just suggest to add `valueOf` implementation for each of them. This will make it easier to use comparison operators 

```
const a = new TuiTime(11, 0);
const b = new TuiTime(12, 0);
If (a < b) {
   doSometing();
}
```

### PR DESCRIPTION
**Pull request by ddubrava**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #1729

## What is the new behavior?

`valueOf` and `Symbol.toPrimitive` are added to TuiDay, TuiMonth, TuiYear and TuiTime

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

**Review by splincode — APPROVED** [review]

File: projects/cdk/date-time/test/day.spec.ts [hunk 1]
```diff
@@ -1046,6 +1046,17 @@ describe('TuiDay', () => {
                     });
                 });
             });
+
+            describe('valueOf returns', () => {
```
Comment (splincode) [inline projects/cdk/date-time/test/day.spec.ts:1050]: I see that e2e tests have fallen, could you add unit tests for the cases that are on the pages?

Comment (ddubrava) [reply projects/cdk/date-time/test/day.spec.ts:1050]: ye, will figure out

**Comment by ddubrava** [comment]
We don't have any integrations for `InputTime`, should I add some? 
An autocomplete there was also broken due to my changes but this bug hasn't been caught.

**Comment by splincode** [comment]
> We don't have any integrations for InputTime, should I add some?

It would be great if you add!

**Comment by splincode** [comment]
@ddubrava Really cool job!

**Review by splincode — APPROVED** [review]

**Review by waterplea — APPROVED** [review]

**Comment by waterplea** [comment]
Thank you!

**Pull request merged**